content/docs/overview.md

```diff
@@ -1792,6 +1792,61 @@ f: Foo(T, N);
 #assert(size_of(f) == (N+N-2)*size_of(T));
 ```
 
+## Custom iterator functions
+
+Custom iterator functions can be used to iterate over anything you want.
+
+A custom iterator function should take a pointer to a struct keeps track of the state of the iteration.
+
+The custom iterator function should return 3 values. The next value to be given to the loop, the next index for the loop, and a boolean to signify if the loop should keep running.
```

Comment:
Custom iterators can return 1, 2, or 3 values, as long as the last value is a boolean.